db/seeds.rb

```diff
@@ -10,6 +10,19 @@
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
 
+# Limit the number of file attachments to add when seeding to reduce time.
+# default limit of 1 can be overriden with env variable 'SEED_ATTACHMENT_PER_SAMPLE'
+@attachments_per_sample = (ENV['SEED_ATTACHMENT_PER_SAMPLE'].presence || 1)
+@attachments_per_sample = @attachments_per_sample.to_i
+# default of unlimited files attached (-1) can be overridden with env variable 'SEED_MAXIMUM_TOTAL_SAMPLE_ATTACHMENTS'
+@maximum_total_sample_attachments = (ENV['SEED_MAXIMUM_TOTAL_SAMPLE_ATTACHMENTS'].presence || -1)
+@maximum_total_sample_attachments = @maximum_total_sample_attachments.to_i
+@total_sample_attachment_count = 0
+# Array of sample file names
+@sequencing_file_list = Rails.root.join('test/fixtures/files').entries.select do |f|
```

Comment:
Since we already know from `@attachments_per_sample` how many files we are going to attach, we should update this to select the first `n` files. You should be able to just add `.first(@attachments_per_sample)` to the `end` on line 24

Reply:
Good idea. I also cleaned up `seed_attachments` function since this change simplifies that loop.